roles/sap_ha_cluster/tasks/main.yml

```diff
@@ -0,0 +1,244 @@
+  vars:
+    __sap_ha_cluster_vars_file: "{{ role_path }}/vars/{{ item }}"
+  when: __sap_ha_cluster_vars_file is file
+
+- name: "SAP HA Cluster - Include common variable construction tasks"
+  ansible.builtin.import_tasks: construct_vars_common.yml
+
+- name: "SAP HA Cluster - Include variable construction for STONITH resources"
+  ansible.builtin.import_tasks: construct_vars_stonith.yml
+
+- name: "SAP HA Cluster - Include variable construction for SAP HANA common"
+  ansible.builtin.include_tasks:
+    file: construct_vars_hana_common.yml
+
+- name: "SAP HA Cluster - Include variable construction for SAP HANA Scale-up"
+  ansible.builtin.include_tasks:
+    file: construct_vars_hana_scaleup.yml
+  when: sap_ha_cluster_sap_type == 'scaleup'
+
+- name: "SAP HA Cluster - Include variable construction for SAP HANA Scale-out"
+  ansible.builtin.include_tasks:
+    file: construct_vars_hana_scaleout.yml
+  when: sap_ha_cluster_sap_type == 'scaleout'
+
+#### for debugging
+# - name: "Fail here for now..."
+#   ansible.builtin.fail:
+
+- name: "SAP HA Cluster - Block with actual changes"
+  block:
+  when:
+    - not sap_ha_cluster_create_config_only
+
+    # Start with prerequisite changes for certain platforms, if applicable
+    - name: "SAP HA Cluster - Include platform specific fencing prerequisites"
+      ansible.builtin.include_tasks:
+        file: "{{ item }}"
+      loop:
+        - "preconfigure_platform_{{ __sap_ha_cluster_platform_type }}.yml"
+      vars:
+        __sap_ha_cluster_platform_file: "{{ role_path }}/tasks/{{ item }}"
+      when:
+        - __sap_ha_cluster_platform_type is defined
+        - __sap_ha_cluster_platform_file is file
+      ignore_errors: "{{ ansible_check_mode }}"
+
+    # Before we are ready to call the ha_cluster role, we want to validate
+    # that the minimum required parameters are defined and not empty.
+    # TODO: make this smarter, currently all these vars are pre-defined anyway
+    - name: "SAP HA Cluster - Validate parameters for 'ha_cluster' role input"
+      ansible.builtin.assert:
+        that:
+          - __sap_ha_cluster_cluster_properties is defined
+          - __sap_ha_cluster_extra_packages is defined
+          - __sap_ha_cluster_resource_primitives is defined
+          - __sap_ha_cluster_resource_clones is defined
+          - __sap_ha_cluster_constraints_order is defined
+          - __sap_ha_cluster_constraints_colocation is defined
+        fail_msg: "Cluster build parameter construction has failed for this parameter."
+
+    - name: "SAP HA Cluster - Query if CIB already exists"
+      ansible.builtin.command:
+        cmd: cibadmin --query
+      register: __sap_ha_cluster_cib_query
+      check_mode: false
+      changed_when: false
+      failed_when: false
+
+    - name: "SAP HA Cluster - Create backup of existing CIB"
+      ansible.builtin.copy:
+        backup: true
+        content: "{{ __sap_ha_cluster_cib_query.stdout }}"
+        dest: /root/cib_backup.xml
```

Comment:
role should and will be executed several time, backup should containt timestamp from my pov.

Reply:
The backup functionality of the copy module is taking care of file naming. It's standard behavior and it is not conflicting the functionality and usability of the backup file for this use case. The backup files will have unique names and the files' mtime will show when they were created.